### ISSUE TITLE
Unable to find Kube Environment associated to the Custom Location

### ISSUE THREAD

**Issue by erjosito** [Azure/logicapps#396]
When following the tutorial at https://docs.microsoft.com/azure/logic-apps/azure-arc-enabled-logic-apps-create-deploy-workflows?tabs=azure-cli, when creating the Logic App the azcli command gave me the error `Unable to find Kube Environment associated to the Custom Location`.

When looking at the extension code, in the function `_get_kube_env_from_custom_location` in `custom.py` seems to be the culprit, since the elements of kube_envs do not seem to have the property `extended_location.custom_location`. If instead you use the property `.additional_properties['extendedLocation']['name']`, the logic app seems to be created successfully.

Here a run of the `az logicapp create` command with the error (plus some additional logging messages I put to inspect the content of the `kube_envs` variable):

```
❯ az logicapp create -n $logicapp_name -g $arc_rg --subscription $subscription_id --storage-account $storage_account_name \
--custom-location $customloc_id --plan $appsvc_plan_name --debug --os-type Linux
cli.knack.cli: Command arguments: ['logicapp', 'create', '-n', 'arclogicapp25568', '-g', 'k8sarc', '--subscription', 'e7da9914-9b05-4891-893c-546cb7b0422e', '--storage-account', 'logicapp24060', '--custom-location', '/subscriptions/e7da9914-9b05-4891-893c-546cb7b0422e/resourcegroups/k8sarc/providers/microsoft.extendedlocation/customlocations/mycluster', '--plan', 'mysvcplan', '--debug', '--os-type', 'Linux']
cli.knack.cli: __init__ debug log:
Enable color in terminal.
cli.knack.cli: Event: Cli.PreExecute []
cli.knack.cli: Event: CommandParser.OnGlobalArgumentsCreate [<function CLILogging.on_global_arguments at 0x7f832094fbf8>, <function OutputProducer.on_global_arguments at 0x7f832087d730>, <function CLI
Query.on_global_arguments at 0x7f832089c840>]
cli.knack.cli: Event: CommandInvoker.OnPreCommandTableCreate []
cli.azure.cli.core: Modules found from index for 'logicapp': ['azext_logicapp']
cli.azure.cli.core: Loading command modules:
cli.azure.cli.core: Name                  Load Time    Groups  Commands
cli.azure.cli.core: Total (0)                 0.000         0         0
cli.azure.cli.core: These extensions are not installed and will be skipped: ['azext_ai_examples', 'azext_next']
cli.azure.cli.core: Loading extensions:
cli.azure.cli.core: Name                  Load Time    Groups  Commands  Directory
cli.azure.cli.core: logicapp                  0.395         5        12  /home/jose/.azure/cliextensions/logicapp
cli.azure.cli.core: Total (1)                 0.395         5        12
cli.azure.cli.core: Loaded 5 groups, 12 commands.
cli.azure.cli.core: Found a match in the command table.
cli.azure.cli.core: Raw command  : logicapp create
cli.azure.cli.core: Command table: logicapp create
cli.knack.cli: Event: CommandInvoker.OnPreCommandTableTruncate [<function AzCliLogging.init_command_file_logging at 0x7f8320325400>]
cli.azure.cli.core.azlogging: metadata file logging enabled - writing logs to '/home/jose/.azure/commands/2021-10-03.23-01-14.logicapp_create.1541.log'.
az_command_data_logger: command args: logicapp create -n {} -g {} --subscription {} --storage-account {} --custom-location {} --plan {} --debug --os-type {}
cli.knack.cli: Event: CommandInvoker.OnPreArgumentLoad [<function register_global_subscription_argument.<locals>.add_subscription_parameter at 0x7f832034b620>, <function register_global_query_examples
_argument.<locals>.register_query_examples at 0x7f83202d28c8>]
cli.knack.cli: Event: CommandInvoker.OnPostArgumentLoad []
cli.knack.cli: Event: CommandInvoker.OnPostCommandTableCreate [<function register_ids_argument.<locals>.add_ids_arguments at 0x7f83202d2950>, <function register_cache_arguments.<locals>.add_cache_argu
ments at 0x7f83202d2a60>]
cli.knack.cli: Event: CommandInvoker.OnCommandTableLoaded []
cli.knack.cli: Event: CommandInvoker.OnPreParseArgs []
cli.knack.cli: Event: CommandInvoker.OnPostParseArgs [<function OutputProducer.handle_output_argument at 0x7f832087d7b8>, <function CLIQuery.handle_query_parameter at 0x7f832089c8c8>, <function regist
er_global_query_examples_argument.<locals>.handle_example_parameter at 0x7f83202d2840>, <function register_ids_argument.<locals>.parse_ids_arguments at 0x7f83202d29d8>]
az_command_data_logger: extension name: logicapp
az_command_data_logger: extension version: 0.1.1
cli.azure.cli.core.commands.client_factory: Getting management service client client_type=WebSiteManagementClient
msrest.universal_http.requests: Configuring retry: max_retries=4, backoff_factor=0.8, max_backoff=90
cli.azure.cli.core.commands.client_factory: Adding custom headers to the client:
cli.azure.cli.core.commands.client_factory:     'x-ms-client-request-id': '0b67e60c-248d-11ec-9fa6-00155d674be3'
cli.azure.cli.core.commands.client_factory:     'CommandName': 'logicapp create'
cli.azure.cli.core.commands.client_factory:     'ParameterSetName': '-n -g --subscription --storage-account --custom-location --plan --debug --os-type'
cli.azure.cli.core.commands.client_factory: Getting management service client client_type=WebSiteManagementClient
msrest.universal_http.requests: Configuring retry: max_retries=4, backoff_factor=0.8, max_backoff=90
cli.azure.cli.core.commands.client_factory: Adding custom headers to the client:
cli.azure.cli.core.commands.client_factory:     'x-ms-client-request-id': '0b67e60c-248d-11ec-9fa6-00155d674be3'
cli.azure.cli.core.commands.client_factory:     'CommandName': 'logicapp create'
cli.azure.cli.core.commands.client_factory:     'ParameterSetName': '-n -g --subscription --storage-account --custom-location --plan --debug --os-type'
cli.azure.cli.core.adal_authentication: AdalAuthentication.signed_session invoked by Track 1 SDK
cli.azure.cli.core._profile: Retrieving token from ADAL for resource 'https://management.core.windows.net/'
cli.azure.cli.core.util: attempting to read file /home/jose/.azure/accessTokens.json as utf-8-sig
adal-python: 95f68b11-c008-47f9-a09f-d069841b2de1 - Authority:Performing instance discovery: ...
adal-python: 95f68b11-c008-47f9-a09f-d069841b2de1 - Authority:Performing static instance discovery
adal-python: 95f68b11-c008-47f9-a09f-d069841b2de1 - Authority:Authority validated via static instance discovery
adal-python: 95f68b11-c008-47f9-a09f-d069841b2de1 - TokenRequest:Getting token from cache with refresh if necessary.
adal-python: 95f68b11-c008-47f9-a09f-d069841b2de1 - CacheDriver:finding with query keys: {'_clientId': '...', 'userId': '...'}
adal-python: 95f68b11-c008-47f9-a09f-d069841b2de1 - CacheDriver:Looking for potential cache entries: {'_clientId': '...', 'userId': '...'}
adal-python: 95f68b11-c008-47f9-a09f-d069841b2de1 - CacheDriver:Found 6 potential entries.
adal-python: 95f68b11-c008-47f9-a09f-d069841b2de1 - CacheDriver:Resource specific token found. 
adal-python: 95f68b11-c008-47f9-a09f-d069841b2de1 - CacheDriver:Returning token from cache lookup, AccessTokenId: b'vuzdC6MYrHZQ/OEErulvEuRGn0b7HLa96W4tYbZakQc=', RefreshTokenId: b'ldwspTg1pWYOvYx3cDIJ+P5r0BAv2DRZ/8CHH6v0ftk='
msrest.http_logger: Request URL: 'https://management.azure.com/subscriptions/e7da9914-9b05-4891-893c-546cb7b0422e/resourceGroups/k8sarc/providers/Microsoft.Web/serverfarms/mysvcplan?api-version=2020-12-01'
msrest.http_logger: Request method: 'GET'
msrest.http_logger: Request headers:
msrest.http_logger:     'Accept': 'application/json'
msrest.http_logger:     'accept-language': 'en-US'
msrest.http_logger:     'User-Agent': 'python/3.6.10 (Linux-5.10.43.3-microsoft-standard-WSL2-x86_64-with-debian-bullseye-sid) msrest/0.6.21 msrest_azure/0.6.3 azure-mgmt-web/2.0.0 Azure-SDK-For-Python AZURECLI/2.28.0 (DEB)'
msrest.http_logger: Request body:
msrest.http_logger: None
msrest.universal_http: Configuring redirects: allow=True, max=30
msrest.universal_http: Configuring request: timeout=100, verify=True, cert=None
msrest.universal_http: Configuring proxies: ''
msrest.universal_http: Evaluate proxies against ENV settings: True
urllib3.connectionpool: Starting new HTTPS connection (1): management.azure.com:443
urllib3.connectionpool: https://management.azure.com:443 "GET /subscriptions/e7da9914-9b05-4891-893c-546cb7b0422e/resourceGroups/k8sarc/providers/Microsoft.Web/serverfarms/mysvcplan?api-version=2020-12-01 HTTP/1.1" 200 None
msrest.http_logger: Response status: 200
msrest.http_logger: Response headers:
msrest.http_logger:     'Cache-Control': 'no-cache'
msrest.http_logger:     'Pragma': 'no-cache'
msrest.http_logger:     'Transfer-Encoding': 'chunked'
msrest.http_logger:     'Content-Type': 'application/json'
msrest.http_logger:     'Content-Encoding': 'gzip'
msrest.http_logger:     'Expires': '-1'
msrest.http_logger:     'Vary': 'Accept-Encoding'
msrest.http_logger:     'Strict-Transport-Security': 'max-age=31536000; includeSubDomains'
msrest.http_logger:     'x-ms-request-id': 'cf270510-85f3-408b-87fc-aa56a0374fe6'
msrest.http_logger:     'Server': 'Microsoft-IIS/10.0'
msrest.http_logger:     'X-AspNet-Version': '4.0.30319'
msrest.http_logger:     'X-Powered-By': 'ASP.NET'
msrest.http_logger:     'x-ms-ratelimit-remaining-subscription-reads': '11999'
msrest.http_logger:     'x-ms-correlation-request-id': '504f4b43-1f5b-4f17-b2c0-27b22214ba99'
msrest.http_logger:     'x-ms-routing-request-id': 'GERMANYWESTCENTRAL:20211003T210116Z:504f4b43-1f5b-4f17-b2c0-27b22214ba99'
msrest.http_logger:     'X-Content-Type-Options': 'nosniff'
msrest.http_logger:     'Date': 'Sun, 03 Oct 2021 21:01:16 GMT'
msrest.http_logger: Response content:
msrest.http_logger: {"extendedLocation":{"name":"/subscriptions/e7da9914-9b05-4891-893c-546cb7b0422e/resourcegroups/k8sarc/providers/microsoft.extendedlocation/customlocations/mycluster","type":"customLocation"},"id":"/subscriptions/e7da9914-9b05-4891-893c-546cb7b0422e/resourceGroups/k8sarc/providers/Microsoft.Web/serverfarms/mysvcplan","name":"mysvcplan","type":"Microsoft.Web/serverfarms","kind":"linux,kubernetes","location":"West Europe","properties":{"serverFarmId":0,"name":"mysvcplan","workerSize":null,"workerSizeId":0,"workerTierName":null,"numberOfWorkers":1,"currentWorkerSize":null,"currentWorkerSizeId":null,"currentNumberOfWorkers":null,"status":"Ready","webSpace":null,"subscription":null,"adminSiteName":null,"hostingEnvironment":null,"hostingEnvironmentProfile":null,"maximumNumbe$OfWorkers":0,"planName":null,"adminRuntimeSiteName":null,"computeMode":"Shared","siteMode":null,"geoRegion":"West Europe","perSiteScaling":true,"elasticScaleEnabled":null,"maximumElasticWorkerCount":null,"numberOfSites":2,"hostingEnvironmentId":null,"isSpot":null,"spotExpirationTime":null,"freeOfferExpirationTime":null,"tags":null,"kind":"linux,kubernetes","resourceGroup":"k8sarc","reserved":true,"isXenon":false,"hyperV":null,"mdmId":null,"targetWorkerCount":null,"targetWorkerSizeId":null,"provisioningState":"Succeeded","webSiteId":null,"existingServerFarmIds":null,"kubeEnvironmentProfile":{"id":"/subscriptions/e7da9914-9b05-4891-893c-546cb7b0422e/resourceGroups/k8sarc/providers/Microsoft.Web/kubeEnvironments/appsvc17937","name":"appsvc17937","type":"Microsoft.Web/kubeEnvironments"},"zoneRedundant":null},"sku":{"name":"K1","tier":"Kubernetes","size":"K1","family":"K","capacity":1}}
cli.azext_logicapp.custom:  ****************** Getting list of kube environments...
cli.azure.cli.core.commands.client_factory: Getting management service client client_type=WebSiteManagementClient
msrest.universal_http.requests: Configuring retry: max_retries=4, backoff_factor=0.8, max_backoff=90
cli.azure.cli.core.commands.client_factory: Adding custom headers to the client:
cli.azure.cli.core.commands.client_factory:     'x-ms-client-request-id': '0b67e60c-248d-11ec-9fa6-00155d674be3' 
cli.azure.cli.core.commands.client_factory:     'CommandName': 'logicapp create'
cli.azure.cli.core.commands.client_factory:     'ParameterSetName': '-n -g --subscription --storage-account --custom-location --plan --debug --os-type'
msrest.async_paging: Paging async iterator protocol is not available for KubeEnvironmentPaged
cli.azure.cli.core.adal_authentication: AdalAuthentication.signed_session invoked by Track 1 SDK
cli.azure.cli.core._profile: Retrieving token from ADAL for resource 'https://management.core.windows.net/'
adal-python: ed3f1474-47c4-44a3-80f0-a795b0166ac5 - Authority:Performing instance discovery: ...
adal-python: ed3f1474-47c4-44a3-80f0-a795b0166ac5 - Authority:Performing static instance discovery
adal-python: ed3f1474-47c4-44a3-80f0-a795b0166ac5 - Authority:Authority validated via static instance discovery
adal-python: ed3f1474-47c4-44a3-80f0-a795b0166ac5 - TokenRequest:Getting token from cache with refresh if necessary.
adal-python: ed3f1474-47c4-44a3-80f0-a795b0166ac5 - CacheDriver:finding with query keys: {'_clientId': '...', 'userId': '...'}
adal-python: ed3f1474-47c4-44a3-80f0-a795b0166ac5 - CacheDriver:Looking for potential cache entries: {'_clientId': '...', 'userId': '...'}
adal-python: ed3f1474-47c4-44a3-80f0-a795b0166ac5 - CacheDriver:Found 6 potential entries.
adal-python: ed3f1474-47c4-44a3-80f0-a795b0166ac5 - CacheDriver:Resource specific token found.
adal-python: ed3f1474-47c4-44a3-80f0-a795b0166ac5 - CacheDriver:Returning token from cache lookup, AccessTokenId: b'vuzdC6MYrHZQ/OEErulvEuRGn0b7HLa96W4tYbZakQc=', RefreshTokenId: b'ldwspTg1pWYOvYx3cDIJ+P5r0BAv2DRZ/8CHH6v0ftk='
msrest.http_logger: Request URL: 'https://management.azure.com/subscriptions/e7da9914-9b05-4891-893c-546cb7b0422e/providers/Microsoft.Web/kubeEnvironments?api-version=2020-12-01'
msrest.http_logger: Request method: 'GET'
msrest.http_logger: Request headers:
msrest.http_logger:     'Accept': 'application/json'
msrest.http_logger:     'accept-language': 'en-US'
msrest.http_logger:     'User-Agent': 'python/3.6.10 (Linux-5.10.43.3-microsoft-standard-WSL2-x86_64-with-debian-bullseye-sid) msrest/0.6.21 msrest_azure/0.6.3 azure-mgmt-web/2.0.0 Azure-SDK-For-Python AZURECLI/2.28.0 (DEB)'
msrest.http_logger: Request body:
msrest.http_logger: None
msrest.universal_http: Configuring redirects: allow=True, max=30
msrest.universal_http: Configuring request: timeout=100, verify=True, cert=None
msrest.universal_http: Configuring proxies: ''
msrest.universal_http: Evaluate proxies against ENV settings: True
urllib3.connectionpool: Starting new HTTPS connection (1): management.azure.com:443
urllib3.connectionpool: https://management.azure.com:443 "GET /subscriptions/e7da9914-9b05-4891-893c-546cb7b0422e/providers/Microsoft.Web/kubeEnvironments?api-version=2020-12-01 HTTP/1.1" 200 None    msrest.http_logger: Response status: 200
msrest.http_logger: Response headers:
msrest.http_logger:     'Cache-Control': 'no-cache'
msrest.http_logger:     'Pragma': 'no-cache'
msrest.http_logger:     'Transfer-Encoding': 'chunked'
msrest.http_logger:     'Content-Type': 'application/json'
msrest.http_logger:     'Content-Encoding': 'gzip'
msrest.http_logger:     'Expires': '-1'
msrest.http_logger:     'Vary': 'Accept-Encoding'
msrest.http_logger:     'Strict-Transport-Security': 'max-age=31536000; includeSubDomains'
msrest.http_logger:     'x-ms-request-id': '0e994e0c-cdcb-40ac-854c-2e5ca4c50968'
msrest.http_logger:     'Server': 'Microsoft-IIS/10.0'
msrest.http_logger:     'X-AspNet-Version': '4.0.30319'
msrest.http_logger:     'X-Powered-By': 'ASP.NET'
msrest.http_logger:     'x-ms-ratelimit-remaining-subscription-reads': '11999'
msrest.http_logger:     'x-ms-correlation-request-id': '7d67a872-60f3-4787-a391-2cc365b03f64'
msrest.http_logger:     'x-ms-routing-request-id': 'GERMANYWESTCENTRAL:20211003T210118Z:7d67a872-60f3-4787-a391-2cc365b03f64'
msrest.http_logger:     'X-Content-Type-Options': 'nosniff'
msrest.http_logger:     'Date': 'Sun, 03 Oct 2021 21:01:17 GMT'
msrest.http_logger: Response content:
msrest.http_logger: {"value":[{"extendedLocation":{"name":"/subscriptions/e7da9914-9b05-4891-893c-546cb7b0422e/resourcegroups/k8sarc/providers/microsoft.extendedlocation/customlocations/mycluster","type":"CustomLocation"},"id":"/subscriptions/e7da9914-9b05-4891-893c-546cb7b0422e/resourceGroups/k8sarc/providers/Microsoft.Web/kubeEnvironments/appsvc17937","name":"appsvc17937","type":"Microsoft.Web/kubeEnvironments","location":"West Europe","tags":{},"properties":{"type":null,"extendedLocation":null,"provisioningState":"Succeeded","deploymentErrors":null,"internalLoadBalancerEnabled":false,"defaultDomain":"appsvc17937-gkrx6y6d2voo.westeurope.k4apps.io","staticIp":"20.86.127.182","arcConfiguration":{"artifactsStorageType":"NetworkFileSystem","artifactStorageClassName":"default","artifactStorageMountPath":null,"artifactStorageNodeName":null,"artifactStorageAccessMode":null,"frontEndServiceConfiguration":{"kind":"LoadBalancer"},"kubeConfig":null},"appLogsConfiguration":null,"workerAppsConfiguration":null,"aksResourceID":null}}],"nextLink":null,"id":null}
cli.azext_logicapp.custom:  ********************* Checking /subscriptions/e7da9914-9b05-4891-893c-546cb7b0422e/resourceGroups/k8sarc/providers/Microsoft.Web/kubeEnvironments/appsvc17937: {'extendedLocation': {'name': '/subscriptions/e7da9914-9b05-4891-893c-546cb7b0422e/resourcegroups/k8sarc/providers/microsoft.extendedlocation/customlocations/mycluster', 'type': 'CustomLocation'}}                 cli.azure.cli.core.util: azure.cli.core.util.handle_exception is called with an exception:
cli.azure.cli.core.util: Traceback (most recent call last):
File "/opt/az/lib/python3.6/site-packages/knack/cli.py", line 231, in invoke
cmd_result = self.invocation.execute(args)
File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 657, in execute
raise ex
File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 720, in _run_jobs_serially
results.append(self._run_job(expanded_arg, cmd_copy))
File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 712, in _run_job
return cmd_copy.exception_handler(ex)
File "/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/appservice/commands.py", line 37, in _ex_handler
raise ex
File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 691, in _run_job
result = cmd_copy(params)
File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 328, in __call__
return self.handler(*args, **kwargs)
File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/command_operation.py", line 121, in handler
return op(**command_args)
File "/home/jose/.azure/cliextensions/logicapp/azext_logicapp/custom.py", line 898, in create_logicapp
if not _validate_asp_and_custom_location_kube_envs_match(cmd, resource_group_name, custom_location, plan):
  File "/home/jose/.azure/cliextensions/logicapp/azext_logicapp/custom.py", line 361, in _validate_asp_and_custom_location_kube_envs_match cmd, custom_location)
File "/home/jose/.azure/cliextensions/logicapp/azext_logicapp/custom.py", line 233, in _get_kube_env_from_custom_location
'Unable to find Kube Environment associated to the Custom Location')
azure.cli.core.azclierror.ResourceNotFoundError: Unable to find Kube Environment associated to the Custom Location
cli.azure.cli.core.azclierror: Unable to find Kube Environment associated to the Custom Location
az_command_data_logger: Unable to find Kube Environment associated to the Custom Location
cli.knack.cli: Event: Cli.PostExecute [<function AzCliLogging.deinit_cmd_metadata_logging at 0x7f8320325620>]
az_command_data_logger: exit code: 3
cli.__main__: Command ran in 4.115 seconds (init: 0.335, invoke: 3.780)
telemetry.save: Save telemetry record of length 3256 in cache
telemetry.check: Negative: The /home/jose/.azure/telemetry.txt was modified at 2021-10-03 22:57:41.815221, which in less than 600.000000 s
```

Not sure if I should create the issue in this repo, I wasn't able to find the correct repo for the CLI extension wheel.

**Comment by ShreeDivyaMV** [comment]
@erjosito, Thanks for bringing this to our attention. We are actively investigating and will get back to you soon.

**Comment by archidda** [comment]
Please upgrade to latest VSIX extension version 0.1.2 for resolution.

You can run these two commands-
az extension remove --name logicapp
az extension add --yes --source "https://aka.ms/logicapp-latest-py2.py3-none-any.whl"

Please note, If a new extension version is available, the current(0.1.2) and later versions show a message saying while this extension is in preview, you can use the following command to upgrade to the latest version without manually removing and installing again:

az logicapp upgrade
